### PR TITLE
Update colors readme with additional definitions

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Adds links to React for higher-order components and hooks.

Additional detail added per comment in #25567 

## How has this been tested?

Confirm links and documentation is accurate.

## Types of changes

Documentation.

### CODE REVIEW

File: packages/block-editor/src/components/colors/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Color HOC and Hook
```
Comment (vcanales) [inline packages/block-editor/src/components/colors/README.md:1]: Perhaps not abbreviating the titles would also be a good idea!

```suggestion
# Color Higher-Order Component and Hook
```

File: packages/block-editor/src/components/colors/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
-Set of functions to enable color functionality in Blocks.
+A set of functions to enable color functionality in blocks. The first two functions use the concept of a higher-order component (HOC), see [React's Higher-Order Components documentation](https://reactjs.org/docs/higher-order-components.html) for additional details.
 
 ## `withColors` HOC
```
Comment (vcanales) [inline packages/block-editor/src/components/colors/README.md:5]:
```suggestion
## `withColors` Higher-Order Component
```

**Review by vcanales — APPROVED** [review]
Thanks for updating! I left two small suggestions.

**Review by aristath — APPROVED** [review]
LGTM! :+1: 

It would be preferable if titles were not abbreviated as per @vcanales's suggestions, but even without that it's a lot better now that there's a reference.
Thank you :)

**Pull request merged**